tools/swift-format/Sources/Rules/NoAccessLevelOnExtensionDeclaration.swift

```diff
@@ -10,7 +10,148 @@ import SwiftSyntax
+    guard let modifiers = node.modifiers, modifiers.count != 0 else { return node }
+    
+    var modifierIterator = modifiers.makeIterator()
+    if let accessKeyword = modifierIterator.next() {
+
+      let keywordKind = accessKeyword.name.tokenKind
+      switch keywordKind {
+      // Public, private, or fileprivate keywords need to be moved to members
+      case .publicKeyword, .privateKeyword, .fileprivateKeyword:
+        diagnose(.moveAccessKeyword(keyword: accessKeyword.name.text), on: accessKeyword)
+        let newMembers = SyntaxFactory.makeMemberDeclBlock(
+          leftBrace: node.members.leftBrace,
+          members: addMemberAccessKeywords(memDeclBlock: node.members, keyword: accessKeyword),
+          rightBrace: node.members.rightBrace)
+        return node.withMembers(newMembers)
+                .withModifiers(removeModifier(curModifiers: modifiers, removal: accessKeyword))
+      // Internal keyword redundant, delete
+      case .internalKeyword:
+        diagnose(.removeRedundantAccessKeyword(name: node.extendedType.description), on: accessKeyword)
+        return node.withModifiers(removeModifier(curModifiers: modifiers, removal: accessKeyword))
+                .withExtensionKeyword(node.extensionKeyword.withOneLeadingNewline())
+      default:
+        return node
+      }
+    }
+    return node
+  }
+  
+  // Returns modifier list without the access modifier
+  func removeModifier(curModifiers: ModifierListSyntax, removal: DeclModifierSyntax) -> ModifierListSyntax {
+    var newMods: [DeclModifierSyntax] = []
+    for modifier in curModifiers {
+      if modifier.name != removal.name {
+        newMods.append(modifier)
+      }
+    }
+    return SyntaxFactory.makeModifierList(newMods)
+  }
+  
+  // Adds given keyword to all members in declaration block
+  func addMemberAccessKeywords(memDeclBlock: MemberDeclBlockSyntax, keyword: DeclModifierSyntax) -> MemberDeclListSyntax {
+    var newMembers: [MemberDeclListItemSyntax] = []
+    let formattedKeyword = keyword.withName(keyword.name.withLeadingTrivia(.newlines(1) + .spaces(2)))
+    
+    for member in memDeclBlock.members {
+      
+      if let varDecl = member.decl as? VariableDeclSyntax {
+        // Check for modifiers, if none, put accessor keyword before var/let keyword
+        guard let modifiers = varDecl.modifiers
+          else {
+            let formattedVarDecl = varDecl.withLetOrVarKeyword(varDecl.letOrVarKeyword.withoutLeadingTrivia())
+            newMembers.append(member.withDecl(formattedVarDecl.addModifier(formattedKeyword)))
+            continue
+          }
+        // If variable already has an accessor keyword, skip (do not overwrite)
+        guard hasAccessorKeyword(modifiers: modifiers) == false
+          else { newMembers.append(member); continue }
+        // Put accessor keyword before the first modifier keyword in the declaration
+        if let next = modifiers.first {
```

Comment:
You might have an easier time building an array of modifiers and then creating a `ModifierListSyntax` at the end, instead of using `appending(_:)` and `replacing(childAt:with:)`.

Also, it might be a bit cleaner if you replace these `if let x = y as? Z` chains with

```swift
switch x {
case let funcDecl as FunctionDeclSyntax:
  // use funcDecl now
default:
  // default
}
```